### DISCUSSION TITLE
API Workflow parameters

### DISCUSSION THREAD

**Discussion by hugo-codestack** [microsoft/farmvibes-ai#115] [Q&A] [answered]
Hi all, 

I have a VM running Farmvibes and the API exposed to be consumed by our product but I have some issues regarding the parameters of the following workflow:

data_processing/index/index 

This is the error message
```
{
  "id": null,
  "location": null,
  "message": "Unable to run workflow with provided parameters. Workflow source raster expects inputs of type <class 'vibe_core.data.rasters.Raster'>, found incompatible types: vibe_core.data.core_types.DataVibe"
}
```

For this I sent this parameters:
```
{
  "name": "test",
  "workflow": "data_processing/index/index",
  "parameters": {
    "index": "ndvi"
  },
  "user_input": {
    "start_date": "2021-02-02",
    "end_date": "2021-08-02",
    "geojson": {
      "type": "Feature",
      "properties": {},
      "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
            [
              -121.802855,
              38.800393
            ],
            [
              -121.803064,
              38.793833
            ],
            [
              -121.797972,
              38.793696
            ],
            [
              -121.798158,
              38.800376
            ],
            [
              -121.802855,
              38.800393
            ]
          ]
        ]
      }
    }
  }
}
```

Can you please help me with this? I want to understand what parameters the workflow expects.

**Comment by rafaspadilha** [comment, marked as answer]
Hi, @hugo-codestack. Thanks for using FarmVibes.AI and raising this question.

You are missing a workflow that downloads a Raster, over which the NDVI will be computed. The workflow `data_processing/index/index` expects a `Raster` object, which comes from a few of our [data_ingestion workflows](https://microsoft.github.io/farmvibes-ai/docfiles/markdown/WORKFLOW_LIST.html#data-ingestion) (e.g., Sentinel-2, LandSat, etc). You are passing as input a `DataVibe` with the geometry and time range, but these should first be used to download the raster.

The following examples and documentation will probably help you:

- [Spectral Indices notebook](https://github.com/microsoft/farmvibes-ai/blob/main/notebooks/sentinel/spectral_indices.ipynb): an example of how to combine a download operation (in this case, Sentinel-2 download) with the `data_processing/index/index` workflow to compute several indices. Specifically, the [indices.yaml](https://github.com/microsoft/farmvibes-ai/blob/main/notebooks/sentinel/indices.yaml) file in this folder shows you a custom workflow that you can adapt for your scenario.
- [Building custom workflow documentation pages](https://microsoft.github.io/farmvibes-ai/docfiles/markdown/WORKFLOWS.html#building-a-workflow) with more details on how to build a custom workflow and combine multiple workflows in FarmVibes.AI.

Please, let me know if you are able to run your workflow.
